agent-inject/agent/annotations.go

```diff
@@ -81,6 +81,11 @@ const (
 	// configuration file and templates can be found.
 	AnnotationAgentConfigMap = "vault.hashicorp.com/agent-configmap"
 
+	// AnnotationAgentGeneratedConfigMapName is the name of the configuration map generated by the
+	// Vault Agent Injector. This is only for internal uses and should not be modified by
+	// users.
+	AnnotationAgentGeneratedConfigMapName = "vault.hashicorp.com/agent-configmap-name"
```

Comment:
If this shouldn't be modified externally, is it possible to not expose this?

Reply:
We add this annotation to the injected pod so we know the name of the configmap generated when we clean it up. We have other annotations (`inject-status`) that are also just used by us. I think this is good to have since a user can check the name of the configmap we've created for them by looking at the annotations. If they change this annotation, then the delete will fail and they will need to clean up the orphaned configmap. I don't see the harm in having this but we could figure out the name of the configmap without it if you think its necessary. This just makes the delete phase much easier.

Reply:
It also occurs to me that setting the configmap name in an annotation on the pod will also make things easier for backwards compatibility if we ever decide to change the configmap naming scheme.

Reply:
Is it common practice for a particular annotation to be exposed, but not expected to be modified? I am just thinking of possible side-effects like orphaned configmaps like you mentioned.

Reply:
Unless I'm missing a way in which this example is different, I believe this is fairly common. The k8s API uses annotations to store metadata like previous configuration and modified time etc.

Reply:
Very much a nit but could a name like `vault.hashicorp.com/agent-configmap-generated-name` make its purpose a bit clearer? It does currently read as though it's an option that should go hand in hand with setting `agent-configmap`.